src/Apps/Order/Routes/__tests__/Review.test.tsx

```diff
@@ -51,124 +56,133 @@ describe("Review", () => {
     )
   }
 
-  it("enables the button and routes to the payoff page", () => {
-    const component = getWrapper(defaultProps)
-    ;(commitMutation as jest.Mock<any>).mockImplementationOnce(
-      (_, { onCompleted }) =>
-        onCompleted({ ecommerceSubmitOrder: { orderOrError: { order: {} } } })
-    )
-    component.find(Button).simulate("click")
-    expect(commitMutation).toHaveBeenCalledTimes(1)
-    expect(pushMock).toBeCalledWith("/orders/1234/status")
+  beforeEach(() => {
+    mutationMock.mockReset()
   })
 
-  it("takes the user back to the /shipping view", () => {
-    const component = getWrapper(defaultProps)
-    component
-      .find(StepSummaryItem)
-      .first()
-      .find("a")
-      .simulate("click")
-    expect(pushMock).toBeCalledWith("/orders/1234/shipping")
-  })
+  describe("buy-mode orders", () => {
+    it("enables the button and routes to the payoff page", () => {
+      const component = getWrapper(defaultProps)
+      mutationMock.mockImplementationOnce((_, { onCompleted }) =>
+        onCompleted({ ecommerceSubmitOrder: { orderOrError: { order: {} } } })
+      )
+      component.find(Button).simulate("click")
+      expect(mutationMock).toHaveBeenCalledTimes(1)
+      expect(pushMock).toBeCalledWith("/orders/1234/status")
+    })
 
-  it("takes the user back to the /payment view", () => {
-    const component = getWrapper(defaultProps)
-    component
-      .find(StepSummaryItem)
-      .last()
-      .find("a")
-      .simulate("click")
-    expect(pushMock).toBeCalledWith("/orders/1234/shipping")
-  })
+    it("takes the user back to the /shipping view", () => {
+      const component = getWrapper(defaultProps)
+      component
+        .find(StepSummaryItem)
+        .first()
+        .find("a")
+        .simulate("click")
+      expect(pushMock).toBeCalledWith("/orders/1234/shipping")
+    })
 
-  it("shows an error modal when there is an error in submitOrderPayload", () => {
-    const component = getWrapper(defaultProps)
+    it("takes the user back to the /payment view", () => {
+      const component = getWrapper(defaultProps)
+      component
+        .find(StepSummaryItem)
+        .last()
+        .find("a")
+        .simulate("click")
+      expect(pushMock).toBeCalledWith("/orders/1234/shipping")
```

Comment:
should this be /payment?

Reply:
Yes! Turned out we weren't resetting the mock. Good catch 🎉